src/text/line.rs

```diff
@@ -451,6 +451,40 @@ impl<'a> Line<'a> {
     pub fn iter_mut(&mut self) -> std::slice::IterMut<Span<'a>> {
         self.spans.iter_mut()
     }
+
+    /// Returns a line that's truncated corresponding to it's alignment and result width
+    #[must_use = "method moves the value of self and returns the modified value"]
```

Comment:
Not something entirely specific to this PR as it’s in other places too…

The ownership of the original object is not moved so the message is confusing with ownership in mind.

It actually does not change the value of self and returns a new, modified value. With a lifetime reference shared with the original.

Not sure how often these must_use messages are misleading in other places too… maybe we should create an issue about it to check that?

Reply:
#995